framework/src/main/java/org/checkerframework/framework/type/QualifierHierarchy.java

```diff
@@ -81,361 +100,351 @@ public int getWidth() {
+     * @param c1 the first collection
+     * @param c2 the second collection
+     * @param result the result collection
      */
-    public Set<? extends AnnotationMirror> greatestLowerBoundsTypeVariable(
-            Collection<? extends AnnotationMirror> annos1,
-            Collection<? extends AnnotationMirror> annos2) {
-        Set<AnnotationMirror> result = AnnotationUtils.createAnnotationSet();
-        for (AnnotationMirror top : getTopAnnotations()) {
-            AnnotationMirror anno1ForTop = null;
-            for (AnnotationMirror anno1 : annos1) {
-                if (isSubtypeTypeVariable(anno1, top)) {
-                    anno1ForTop = anno1;
-                }
-            }
-            AnnotationMirror anno2ForTop = null;
-            for (AnnotationMirror anno2 : annos2) {
-                if (isSubtypeTypeVariable(anno2, top)) {
-                    anno2ForTop = anno2;
-                }
-            }
-            AnnotationMirror t = greatestLowerBoundTypeVariable(anno1ForTop, anno2ForTop);
-            if (t != null) {
-                result.add(t);
-            }
+    static void assertSameSize(Collection<?> c1, Collection<?> c2, Collection<?> result) {
+        if (c1.size() != result.size()) {
+            throw new BugInCF(
+                    "inconsistent sizes (%d, %d, %d):%n  %s%n  %s%n  %s",
+                    c1.size(), c2.size(), result.size(), c1, c2, result);
         }
-        return result;
     }
 
+    // **********************************************************************
+    // Deprecated methods
+    // **********************************************************************
+
     /**
-     * Returns true if and only if the given type can have empty annotation sets (and thus the
-     * *TypeVariable methods need to be used).
-     */
-    public static boolean canHaveEmptyAnnotationSet(AnnotatedTypeMirror type) {
-        return type.getKind() == TypeKind.TYPEVAR
-                || type.getKind() == TypeKind.WILDCARD
-                ||
-                // TODO: or should the union/intersection be the LUB of the alternatives?
-                type.getKind() == TypeKind.UNION
-                || type.getKind() == TypeKind.INTERSECTION;
+     * Tests whether {@code subQualifier} is a sub-qualifier of, or equal to, {@code
+     * superQualifier}, according to the type qualifier hierarchy.
+     *
+     * <p>This method works even if the underlying Java type is a type variable. In that case, a
+     * 'null' AnnotationMirror is a legal argument that represents no annotation.
+     *
+     * @param subQualifier a qualifier that might be a subtype
+     * @param superQualifier a qualifier that might be a subtype
+     * @return true iff {@code subQualifier} is a subqualifier of, or equal to, {@code
+     *     superQualifier}
+     * @deprecated Without the bounds of the type variable, it is not possible to correctly compute
```

Comment:
For all the deprecated methods, can you point to the methods that should be used instead?